### ISSUE TITLE
Property noLayout does not exist on Route when implementing custom routes (with Typescript)

### ISSUE THREAD

**Issue by ecorrales1979** [marmelab/react-admin#6126]
I'm trying to make a forgot password page, which need to be rendered without the layout. I'm following the example in documentation, Admin page, customRoutes section and adapting it to Typescript. Without the noLayout property, the route is rendering, but within the layout. The problem arises when adding noLayout to the route:

Property 'noLayout' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Route<{}, "/forgot-password">> & Readonly<RouteProps<"/forgot-password", {}> & OmitNative<...>> & Readonly<...>'.

I realized that I'm using a Route component from react-router-dom implementing RouterProps and the property noLayout is declared in CustomRoute (extending RouterProps). The Admin's customRoutes property is waiting for an array of something implementing CustomRoutes interface, but instead I'm providing an array of Routes (like it is in the docs)

Here is my App.tsx:
```js
import React from 'react'
import { Admin, Resource } from 'react-admin'
import { Route } from 'react-router-dom'
import portugueseMessages from '@henriko/ra-language-portuguese'
import polyglotI18nProvider from 'ra-i18n-polyglot'

import dataProvider from './providers/dataProvider'
import authProvider from './providers/authProvider'
import Layout from './layout/Layout'
import Login from './pages/Login'
import Dashboard from './pages/Dashboard'
import ForgotPassword from './pages/ForgotPassword'

const App: React.FC = () => {
  const i18nProvider = polyglotI18nProvider(() => portugueseMessages, 'pt')

  return (
    <Admin
      dataProvider={dataProvider}
      authProvider={authProvider}
      i18nProvider={i18nProvider}
      layout={Layout}
      loginPage={Login}
      customRoutes={[
        <Route
          key="forgot-password"
          path="/forgot-password"
          component={ForgotPassword}
          exact
          noLayout
        />
      ]}
    >
      <Resource {...Dashboard} />
    </Admin>
  )
}

export default App
```

**Comment by djhi** [comment]
We already export a `CustomRoute` interface from `react-admin`. You can use it like this:

```tsx
import React from 'react'
import { Admin, Resource, CustomRoute } from 'react-admin'
import { Route } from 'react-router-dom'
import portugueseMessages from '@henriko/ra-language-portuguese'
import polyglotI18nProvider from 'ra-i18n-polyglot'

import dataProvider from './providers/dataProvider'
import authProvider from './providers/authProvider'
import Layout from './layout/Layout'
import Login from './pages/Login'
import Dashboard from './pages/Dashboard'
import ForgotPassword from './pages/ForgotPassword'

const App: React.FC = () => {
  const i18nProvider = polyglotI18nProvider(() => portugueseMessages, 'pt')

  return (
    <Admin
      dataProvider={dataProvider}
      authProvider={authProvider}
      i18nProvider={i18nProvider}
      layout={Layout}
      loginPage={Login}
      customRoutes={[
        <Route<CustomRoute>
          key="forgot-password"
          path="/forgot-password"
          component={ForgotPassword}
          exact
          noLayout
        />
      ]}
    >
      <Resource {...Dashboard} />
    </Admin>
  )
}

export default App
```

**Comment by fzaninotto** [comment]
i believe this should be documented, though.

**Comment by fzaninotto** [comment]
Fixed by #6158
